src/imageResource.js

```diff
@@ -80,7 +78,7 @@ ImageResource.prototype = {
             self.visible = true;
             self.status = 'shown';
             self.parent.viewer.removeHandler('tile-drawn', tileDrawnHandler);
-            self.dispatcher.emit('image-resource-tile-source-opened', { 'detail': self.tileSource });
+            self.parent.dispatcher.emit('image-resource-tile-source-opened', { 'detail': self.tileSource });
```

Comment:
You do have a `self.dispatcher`, so you don't need to use `self.parent.dispatcher`. Of course yours is just copied from theirs, but I think that's better to do it once than throughout the code...that way if we change to a different dispatcher it's an easier change.